plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/clustering/TestClusteringMetadataUtil.java

```diff
@@ -0,0 +1,822 @@
+
+    private static class TableSnapShotBuilderForTestBuilder
+    {
+        private String tableLocation;
+
+        public TableSnapShotBuilderForTestBuilder withTableLocation(String tableLocation)
+        {
+            this.tableLocation = tableLocation;
+            return this;
+        }
+
+        public TableSnapshot build()
+                throws IOException
+        {
+            return load(
+                    SESSION,
+                    new FileSystemTransactionLogReader(tableLocation, CREDENTIALS_HANDLE, DEFAULT_DELTA_LAKE_FILE_SYSTEM_FACTORY),
+                    new SchemaTableName("schema", "test"),
+                    Optional.empty(),
+                    tableLocation,
+                    ParquetReaderOptions.defaultOptions(),
+                    true,
+                    DOMAIN_COMPACTION_THRESHOLD,
+                    DEFAULT_TRANSACTION_LOG_MAX_CACHED_SIZE,
+                    Optional.empty());
+        }
+    }
+
+    private static class CommitInfoEntryForTestBuilder
+    {
+        private String operation;
+        private Map<String, String> operationParameters;
+
+        public CommitInfoEntryForTestBuilder withOperation(String operation)
+        {
+            this.operation = operation;
+            return this;
+        }
+
+        public CommitInfoEntryForTestBuilder withOperationParameters(Map<String, String> operationParameters)
+        {
+            this.operationParameters = operationParameters;
+            return this;
+        }
+
+        public CommitInfoEntry build()
+        {
+            requireNonNull(operation, "operation is null");
+            requireNonNull(operationParameters, "operationParameters is null");
+            long timestamp = Instant.now().toEpochMilli();
+            return new CommitInfoEntry(
+                    0,
+                    OptionalLong.of(timestamp),
+                    timestamp,
+                    "user1",
+                    "user1",
+                    operation,
+                    ImmutableMap.copyOf(operationParameters),
+                    null,
+                    null,
+                    "cluster_1",
+                    0,
+                    "SnapshotIsolation",
+                    Optional.of(true),
+                    ImmutableMap.of());
+        }
+    }
+
+    @Test
+    void testGetLatestClusteredColumnsWithAutoOptimize()
+            throws IOException
+    {
+        String tableLocation = "databricks172/liquid_clustering_with_operations/liquid_clustering_with_auto_optimize";
+        TableSnapshot tableSnapshot = new TableSnapShotBuilderForTestBuilder()
+                .withTableLocation(getResourceLocation(tableLocation))
+                .build();
+
+        Optional<List<String>> result = ClusteringMetadataUtil.getLatestClusteredColumns(FILE_SYSTEM, tableSnapshot);
+
+        assertThat(result.isPresent()).isTrue();
+        assertThat(ImmutableList.of()).isEqualTo(result.get());
```

Comment:
AssertJ provides contains method for optional type.
```java
assertThat(result).contains(ImmutableList.of());
```

Also, the order of parameter is wrong. `assertThat()` should take actual, not expected.